app/models/concerns/issues/canonical_data.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module Issues::CanonicalData
+  extend ActiveSupport::Concern
+
+  included do
+    before_save :set_canonical_data, if: :data_changed?
+  end
+
+  private
+
+    def set_canonical_data
+      self.canonical_data = (data_type == 'single_row' ? converted_data.first : nil)
```

Comment:
Quizás acá está el tema, fijate con un:
```suggestion
      self.canonical_data = data_type == 'single_row' ? converted_data.first.to_json : nil
```

Reply:
claro